### DISCUSSION TITLE
qPDF running with PHP

### DISCUSSION THREAD

**Discussion by roalvesrj** [qpdf/qpdf#503] [Q&A]
Dear,

I use [PDFParse](https://github.com/smalot/pdfparser) to read the contents of bank files in PDF and until some time ago it ran normally, however, the version of PDF used by the bank, went to 1.5 and with that I can't read anymore. I saw that using qPDF I can "correct" the file and it is read by PDFParse normally. I would like to know how can I do to run qPDF with PHP so that it can fix all my files loaded before reading?

Thanks

**Comment by jberkenbilt** [comment]
Hello! qpdf has no specific support for php, so your best bet here would be to have your php code just run qpdf from the command line. Just preprocess the file with qpdf into a temporary file and then feed that to to PDFParse.

**Comment by jberkenbilt** [comment]
Regarding PDFParse not accepting 1.5 files, I can only guess what the issue might be. Perhaps it doesn't support 256-bit encryption, or perhaps it doesn't support object streams, or maybe it just doesn't like the version number. If you invoke qpdf with `--force-version=1.4`, it will set the version header to 1.4 and, if it detects encryption that is not supported in 1.4, it will decrypt the file automatically. It will also disable object streams. However, qpdf does not look at document structure or content stream data for version compatibility, so it's conceivable that the file may still contain some constructs that were not introduced until 1.5.

All that said, my suggestion would be to run `qpdf --force-version=1.4 yourfile.pdf tmpfile.pdf` and then pass tmpfile.pdf to PDFParse. More likely than not, that will solve your problem.
